singer_sdk/tap_base.py

```diff
@@ -379,6 +379,9 @@ def sync_all(self) -> None:
             stream.sync()
             stream.finalize_state_progress_markers()
 
+        for stream in self.streams.values():
+            stream.log_sync_costs()
+
```

Comment:
Wdyt about moving this to the first loop?

```suggestion
            stream.sync()
            stream.finalize_state_progress_markers()
            stream.log_sync_costs()
```

Reply:
The only reason I did it this way was so all logging would be grouped at the end of the tap run, and not sprinkled around.
Since I mostly grep logs, it doesn't actually matter too much, now that I think of it.
Code would look cleaner without this. Your call :)

Reply:
@laurentS Yeah, if you don't mind too much and still works for the use case you have in mind, let's keep the single loop 🙂

Reply:
Well, I messed up here by replying without properly looking at the code. There was another reason that I used 2 loops: the first loop skips all child streams, so the logging only happens with the top level stream (this is what I'm seeing in our logs now).
My bad. I will open a separate PR to revert this change, we do need 2 loops in the end.

Reply:
The fix is in #752. @edgarrmondragon sorry for messing this up.